src/lib.rs

```diff
@@ -598,7 +601,8 @@ impl IntoIter {
         // Make room for another open file descriptor if we've hit the max.
         if self.stack_list.len() - self.oldest_opened == self.opts.max_open {
             self.stack_list[self.oldest_opened].close();
-            self.oldest_opened = self.oldest_opened.checked_add(1).unwrap();
+            // Unwrap is safe here as long as we assume that `self.opts.max_open` is never 0
```

Comment:
Upon further thought, is there a reason this was made a checked_add rather than just using `+ 1`?